src/lib.rs

```diff
@@ -95,13 +95,18 @@ pub fn create_host(book: Arc<Book>, labels: Arc<Labels>, compile_opts: CompileOp
 pub fn check_book(book: &mut Book) -> Result<(), Info> {
   // TODO: Do the checks without having to do full compilation
   // TODO: Shouldn't the check mode show warnings?
-  compile_book(book, CompileOpts::light())?;
+  compile_book(book, None, CompileOpts::light())?;
   Ok(())
 }
 
-pub fn compile_book(book: &mut Book, opts: CompileOpts) -> Result<CompileResult, Info> {
-  let warns = desugar_book(book, opts)?;
+pub fn compile_book(
+  book: &mut Book,
+  args: Option<Vec<Term>>,
```

Comment:
For compiling programs that receive arguments, we want to compile down to an hvm-core program that receives arguments instead of passing the arguments as compile-time constants like this.

For running programs with hvm-lang the current approach is good